Otherwise we fail as:

[cweyl@athena Catalyst-Model-XMLRPC-0.04]$ XMLRPC_TEST_LIVE=1 make test
PERL_DL_NONLAZY=1 /usr/bin/perl "-MExtUtils::Command::MM" "-e" "test_harness(0, 'inc', 'blib/lib', 'blib/arch')" t/*.t
t/01use............ok                                                        
t/02pod............skipped
        all skipped: set TEST_POD to enable this test
t/03podcoverage....skipped
        all skipped: set TEST_POD to enable this test
t/04execute........Can't use an undefined value as a HASH reference at /usr/lib/perl5/vendor_perl/5.10.0/Catalyst/Component.pm line 93.
# Looks like your test died before it could output anything.
t/04execute........dubious                                                   
        Test returned status 255 (wstat 65280, 0xff00)
DIED. FAILED tests 1-3
        Failed 3/3 tests, 0.00% okay
Failed Test   Stat Wstat Total Fail  List of Failed
-------------------------------------------------------------------------------
t/04execute.t  255 65280     3    6  1-3
2 tests skipped.
Failed 1/4 test scripts. 3/5 subtests failed.
Files=4, Tests=5,  1 wallclock secs ( 0.28 cusr +  0.04 csys =  0.32 CPU)
Failed 1/4 test programs. 3/5 subtests failed.

diff -ur Catalyst-Model-XMLRPC-0.04.orig/lib/Catalyst/Model/XMLRPC.pm Catalyst-Model-XMLRPC-0.04/lib/Catalyst/Model/XMLRPC.pm
--- Catalyst-Model-XMLRPC-0.04.orig/lib/Catalyst/Model/XMLRPC.pm	2007-07-04 03:30:24.000000000 -0700
+++ Catalyst-Model-XMLRPC-0.04/lib/Catalyst/Model/XMLRPC.pm	2008-09-18 08:20:36.314621277 -0700
@@ -16,7 +16,7 @@
     my ($class, $c, $config) = @_;
 
     my $self = $class->NEXT::new($c, $config);
-    $self->config($config);
+    $self->config($config) if defined $config;
 
     return $self;
 }
